zoo/analytics/tasks/utils.py

```diff
@@ -59,3 +61,64 @@ def resolve_unknown_format(version: str) -> str:
         else:
             break
     return ".".join(temp_version)
+
+
+class DockerImageId:
+    """Represents parsed docker image id.
+
+    TODO: move this to some more appropriate OSS package.
+    """
+
+    # pylint: disable=too-many-instance-attributes # Ignore PyCommentedCodeBear
+
+    def __init__(self, image: str):  # Ignore RadonBear
+        self.full_image_name = image
+
+        self.registry = None
+        self.port = None
+        self.username = None
+        self.repository_name = None
+        self.namespaces = None
+        self.tag = None
+        self.version = None
+        self.os = None
+        self.os_version = None
+        self.alias = None
+
+        if "/" in image:
+            before, _, _ = image.partition("/")
+            if "localhost" in before or "." in before:
+                self.registry, _, image = image.partition("/")
+
+                if ":" in self.registry:
+                    self.registry, _, self.port = self.registry.partition(":")
+
+        if ":" in image:
+            image, _, self.tag = image.partition(":")
+
+        if "/" in image:
+            self.username, _, self.repository_name = image.partition("/")
+        else:
+            self.repository_name = image
+
+        if "/" in self.repository_name:
+            self.repository_name, _, self.namespaces = self.repository_name.partition(
+                "/"
+            )
+            self.namespaces = self.namespaces.split("/")
+
+        if self.tag and " as " in self.tag:
+            self.tag, _, self.alias = self.tag.partition(" as ")
+
+        if self.tag and "-" in self.tag:
+            self.version, _, self.os = self.tag.partition("-")
+
+            match = re.search(r"\d+[\d|\.]*\d", self.os)
+            if match:
+                self.os_version = match.group()
+                self.os = self.os.replace(self.os_version, "")
+        else:
+            self.version = self.tag
+
+    def __str__(self):
+        return str(self.__dict__)
```

Comment:
This would actually print all the stuff that object has, right? i think we should print only the attributes we parse here.

Reply:
No, `__dict__` on an instance returns attributes only, so works ok in this case.